✨ Add extraArgs option

zaber-paul opened the pull request:
## Description

This can be used to provide extra arguments to the 'git commit' command, e.g. '--fixup HEAD'

## Tests

<!-- Ensure that all the tests passed -->
Snapshot was updated with `yarn jest --updateSnapshot`
- [x] All tests passed.

carloscuesta commented on src/prompts.js:
```diff
@@ -67,6 +67,11 @@ const gitmoji = (gitmojis) => {
+    {
+      name: 'extraArgs',
+      message: 'Extra commit args (e.g. --fixup HEAD):',
+      validate: guard.extraArgs
```
The `guard.extraArgs` method does not exists in the `guard.js` file, maybe you forgot to add it?

carloscuesta commented on test/__snapshots__/gitmojiCli.spec.js.snap:
```diff
@@ -132,6 +132,11 @@ Array [
+  Object {
+    "message": "Extra commit args (e.g. --fixup HEAD):",
+    "name": "extraArgs",
+    "validate": undefined,
```
Referring to the comment I posted, that's why this is undefined in the snapshot

zaber-paul commented:
I removed the validation function since any options (and future git commit options) should be allowed. If you enter incorrect text for `extraArgs`, the commit will fail with a message from `git`. Also, updated the snapshots.

I only ever use the `--fixup` option here (personally), so I'd be OK to rewrite this PR as a dedicated prompt for fixup reference if you prefer (it could have a validation function matching SHAs and branch names). Or, if you prefer to merge I think this PR is in a mergeable state now.

carloscuesta commented:
Hey @zaber-paul 

Thanks for the work you put into this PR, I really appreciate that❤️ 

I want to focus on making the cli simpler and not introduce a lot of new features.

Thanks!
